### ISSUE TITLE
[📑 Docs]: Provide content for `AsyncAPI Context` document

### ISSUE THREAD

**Issue by derberg** [asyncapi/generator#808]
### What Dev Docs changes are you proposing?

https://github.com/asyncapi/generator/blob/master/docs/asyncapi-context.md

Actually, we need to call it Template Context.

Scope, pretty much copy paste from https://github.com/asyncapi/template-for-generator-templates#asyncapi-document. Of course if you see it could be somehow improved, please do not hesitate!

Maybe what could be added is more info about parameters :thinkin: up to you. If while writing you will think you still don't know what parameters are, then let us brainstorm.

In short params allow template developer to enable customization of the template output, basing on different paramaters, like for example `javaPackage` param that is used here https://github.com/asyncapi/java-spring-template to enable users to specify before generation, what is the name of java package for generated classes. 

cc @pratik2315 @Florence-Njeri

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by derberg** [comment]
solved by @Florence-Njeri 

https://github.com/asyncapi/generator/blob/master/docs/template-context.md
